solana/solana-ibc/programs/solana-ibc/src/chain.rs

```diff
@@ -173,6 +174,42 @@ impl ChainData {
             .map_err(into_error)
     }
 
+    /// Updates multiple validators stake based on the operation
+    ///
+    /// Fails if candidate is not found during remove operation
+    /// or if the subtraction overflows
+    pub fn set_stake_multiple(
+        &mut self,
+        validator_change_in_stake: Vec<(Pubkey, u128)>,
+        operation: StakeOperation,
```

Comment:
I think we can get away with having i128 in the map and then we don’t need StakeOperation.